api/src/graph.rs

```diff
@@ -53,7 +53,7 @@ pub trait Graph {
     where
         Self: 'x;
     /// The error type that this graph may raise.
-    type Error: Error + 'static;
+    type Error: Error + 'static + Send + Sync;
```

Comment:
Knowing that this _is_ a breaking change, I held off on explicitly feature flagging unless explicitly requested. This can easily be modified:

```suggestion

    #[cfg(not(feature = "threadsafe_errs"))]
    type Error: Error + 'static;
    #[cfg(feature = "threadsafe_errs")]
    type Error: Error + 'static + Send + Sync;
```